### PR TITLE
Answer:18

### PR DESCRIPTION
**Pull request by BruceArmstrong007**
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

Added testcases with cypress and testing library

### CODE REVIEW

File: apps/testing/nested/src/app/child.component.cy.ts [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
   });
 
   describe('When typing "Good" and clicking on Validate', () => {
     test('Then show "Title is Good" message, no error message and send a http request to the backend', async () => {
       setup();
+      cy.get('input[type=text]').clear();
+      cy.get('input[type=text]').type('Good');
+      cy.get('button').click();
+      cy.contains('Title is required !!!').should('not.exist');
+      cy.get('@http').should('not.called');
```
Comment (tomalaforge) [inline apps/testing/nested/src/app/child.component.cy.ts:31]: are you sure ? Http should be called

File: apps/testing/nested/src/app/child.component.spec.ts [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
+import { createMockWithValues } from '@testing-library/angular/jest-utils';
+import { HttpService } from './http.service';
 
 describe('ChildComponent', () => {
+  const setup = async () => {
+    const httpServiceMock = createMockWithValues(HttpService, {
+      sendTitle: jest.fn(),
+    });
+    await render(ChildComponent, {
+      componentProviders: [
```
Comment (tomalaforge) [inline apps/testing/nested/src/app/child.component.spec.ts:13]: it should be on the provider property

componentProviders is only to override service provider at component level

File: apps/testing/nested/src/app/child.component.spec.ts [hunk 3]
```diff
@@ -1,16 +1,50 @@
...
+        {
+          provide: HttpService,
+          useValue: httpServiceMock,
```
Comment (tomalaforge) [inline apps/testing/nested/src/app/child.component.spec.ts:16]: it's nicer this way, why didn't you do it the same way with Cypress ?

File: apps/testing/nested/src/app/child.component.spec.ts (hunk 2 above)
Comment (BruceArmstrong007) [reply apps/testing/nested/src/app/child.component.spec.ts]: i have changed this to providers

**Pull request closed without merging**